containers_test.go

```diff
@@ -30,6 +30,68 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+func TestExecExitError(t *testing.T) {
+	ctx := testlog.WithTB(context.Background(), t)
+	client := DockerClient()
+	containerID, err := CreateContainer(ctx, client, os.Stdout, &ContainerDefinition{
+		Label:   "narwhal-exec-exit-error",
+		Image:   "yourbase/yb_ubuntu:18.04",
+		Command: "/usr/bin/tail -f /dev/null",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		err := client.StopContainerWithContext(containerID, 2, ctx)
+		if err != nil {
+			t.Logf("stopping container %s: %v", containerID, err)
+		}
+		err = client.RemoveContainer(docker.RemoveContainerOptions{
+			ID:      containerID,
+			Context: ctx,
+		})
+		if err != nil {
+			t.Logf("removing container %s: %v", containerID, err)
+		}
+	}()
+
+	err = StartContainer(ctx, client, containerID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if running, err := IsRunning(ctx, client, containerID); !running {
+		t.Fatalf("Container %s didn't start, but should have, as StartContainer was called:\n%v", containerID, err)
+	}
+
+	cmdStat := "stat /somewhere/strange/that/do/not/exist"
+	err = ExecShell(ctx, client, containerID, cmdStat, &ExecShellOptions{
+		CombinedOutput: os.Stdout,
+	})
+	if err != nil {
+		if code, ok := IsExitError(err); !ok || code != 1 {
+			t.Errorf("error = %v; want exit code 1", err)
+		}
+	} else {
+		t.Error("Expecting an error, but none was returned")
+	}
```

Comment:
The `err != nil` check is not necessary: `IsExitError` is sufficient to check.

```suggestion
	if code, ok := IsExitError(err); !ok || code != 1 {
		t.Errorf("error = %v; want exit code 1", err)
	}
```

Reply:
If err is `nil`, `IsExitError` won't be able to treat anything, so no, it should be kept as is. When the test was failing (without a `CombinedOutput`), it was showing "Expecting an error, but none was returned".

Reply:
I don't understand the objection. If I run:

```go
package main

import (
	"fmt"
	"github.com/yourbase/narwhal"
)

func main() {
	fmt.Println(narwhal.IsExitError(nil))
}
```

The output is:

```
0 false
```

So `!ok || code != 1` would be `true`, and thus trigger a test failure.

Reply:
It gave me an error, locally. Maybe I did something wrong. I'll check back